compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -123,21 +121,26 @@ fn build_fixed_size_array_di_node<'ll, 'tcx>(
 
     let (size, align) = cx.size_and_align_of(array_type);
 
-    let upper_bound = len
-        .try_to_target_usize(cx.tcx)
-        .expect("expected monomorphic const in codegen") as c_longlong;
+    let upper_bound =
+        len.try_to_target_usize(cx.tcx).expect("expected monomorphic const in codegen");
 
-    let subrange =
-        unsafe { Some(llvm::LLVMRustDIBuilderGetOrCreateSubrange(DIB(cx), 0, upper_bound)) };
+    let subrange = unsafe {
+        llvm::LLVMDIBuilderGetOrCreateSubrange(
+            DIB(cx),
+            /* LowerBound */ 0i64,
+            /* Count */ upper_bound as i64,
```

Comment:
These have to be positive numbers, right?

Reply:
LLVM declares them as `int64_t`, in both the C and C++ APIs. 🤷‍♂️

Or are you suggesting that we assert that they're non-negative?

Reply:
I am indeed suggesting that.